### PR TITLE
docs: Describe data capture configuration

### PR DESCRIPTION
**Pull request by piob-io**
This change is [Reviewable](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3)

### CODE REVIEW

**Comment by piob-io** [comment]
* **#3** <a href="https://app.graphite.dev/github/pr/DataExMachina-dev/side-eye-public/3?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @piob-io and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/data-capture.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+title: "Data capture"
+weight: 30
+---
+
+# Data capture
+
+Side-eye data capture operates by navigating the memory, starting from frames -
+i.e. currently executed functions, following nested structures and pointers.
```
Comment (ajwerner) [inline docs/content/data-capture.md]: nit: how about `currently executing functions`? Something about `currently` with the past tense seems off.

File: docs/content/data-capture.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+To limit the amount of captured data, only configured functions and expressions
+are included. Expressions are simply paths to reach specific variable, following
+nesting and pointers. Expressions can reference primitive types or structures
+(captured as a whole), or more complex types with custom handling.
+
+## Complex types
+
+### Interfaces and any values
```
Comment (ajwerner) [inline docs/content/data-capture.md]: Nit: put backticks on `any`?

File: docs/content/data-capture.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+When expression references an interface, side-eye will unbox it to discover and
```
Comment (ajwerner) [inline docs/content/data-capture.md]: nit: "When an expression"

**Review by ajwerner — APPROVED** [review]
LGTM, just nits

File: docs/content/data-capture.md (hunk 1 above)
Comment (piob-io) [reply docs/content/data-capture.md]: done

File: docs/content/data-capture.md (hunk 2 above)
Comment (piob-io) [reply docs/content/data-capture.md]: done

File: docs/content/data-capture.md (hunk 3 above)
Comment (piob-io) [reply docs/content/data-capture.md]: done

**Review by andreimatei — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3)* status: 0 of 2 files reviewed, 7 unresolved discussions (waiting on @ajwerner and @piob-io)

___
*[`docs/content/data-capture.md` line 8 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#-O9zsiuA6Gk1nXNkPOfs:-O9zsiuA6Gk1nXNkPOft:b2w3yjz) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L8)):*
> ```Markdown
> # Data capture
> 
> Side-eye data capture operates by navigating the memory, starting from frames -
> ```

I would spell you _when_ we capture data - for snapshots, when we walk the stacks of goroutines, and for events, when the instrumented/probed function executes.
___
*[`docs/content/data-capture.md` line 8 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#-O9zz5tm98jwbPUf1hEO:-O9zz5tm98jwbPUf1hEP:bbk4626) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L8)):*
> ```Markdown
> # Data capture
> 
> Side-eye data capture operates by navigating the memory, starting from frames -
> ```

s/Side-eye/Side-Eye
___
*[`docs/content/data-capture.md` line 12 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#-O9zyBLU7i24kUHjKWu1:-O9zyBLU7i24kUHjKWu2:bk3qurc) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L12)):*
> ```Markdown
> 
> To limit the amount of captured data, only configured functions and expressions
> are included. Expressions are simply paths to reach specific variable, following
> ```

to reach _a_ specific variable?

I would expand this paragraph to introduce the notion of a spec
___
*[`docs/content/data-capture.md` line 18 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#gh-1811339345:-O9ztNbJ6VWk-gcOgC_P:bxlajm9) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L18)):*
<details><summary><i>Previously, piob-io wrote…</i></summary><blockquote>

done
</blockquote></details>

`any` is simply `interface{}`. I wouldn't talk about "any value", I don't think that's a common Go thing to say.
___
*[`docs/content/data-capture.md` line 20 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#gh-1811340213:-O9zuDDaDRuArgOwcrS1:b-7rl0iw) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L20)):*
<details><summary><i>Previously, piob-io wrote…</i></summary><blockquote>

done
</blockquote></details>

done?
___
*[`docs/content/data-capture.md` line 20 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#-O9zuFN1EBV74bRhP7VS:-O9zuFN1EBV74bRhP7VT:bhg7kjy) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L20)):*
> ```Markdown
> ### Interfaces and any values
> 
> When expression references an interface, side-eye will unbox it to discover and
> ```

s/side-eye/Side-Eye
___
*[`docs/content/data-capture.md` line 21 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#-O9zuIBE5EL9ME9oXX_J:-O9zuIBE5EL9ME9oXX_K:b-zcrzmq) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L21)):*
> ```Markdown
> 
> When expression references an interface, side-eye will unbox it to discover and
> report underlying object type. Additionally, types themselves can be configured
> ```

_the_ underlying object type.
Better yet, let's say "the dynamic type of the value boxed in the interface".
___
*[`docs/content/data-capture.md` line 22 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#-O9zuV2sBxuDQlPKtjaS:-O9zuV2sBxuDQlPKtjaT:b-1hmmte) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L22)):*
> types can be configured... or to capture them as a show

Something doesn't sound right.
___
*[`docs/content/data-capture.md` line 62 at r2](https://reviewable.io/reviews/DataExMachina-dev/side-eye-public/3#-O9zumqfAgXYDU5lCQ8B:-O9zumqfAgXYDU5lCQ8C:b-1f623m) ([raw file](https://github.com/DataExMachina-dev/side-eye-public/blob/501388ff614aeb0656e7b7877d770e2a35b6b124/docs/content/data-capture.md#L62)):*
> ```Markdown
>   (as with regular interfaces)
> 
> Note that only types of keys and values are compared for matching, not the data
> ```

_the_ types of keys

**Comment by piob-io** [comment]
Done - with the exception of introducing a paragraph for the nomenclature.
Thanks!

**Pull request merged**